### PR TITLE
Prepare for v0.4 release for  user_events exporters for metrics and logs

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

New releases for user_events exporters with latest otel-rust (v0.23.0) dependency.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-stackdriver/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ prost = "0.12"
 prost-types = "0.12"
 thiserror = "1.0.30"
 tonic = { version = "0.11", features = ["gzip", "tls", "transport"] }
-yup-oauth2 = { version = "8.3.3", optional = true }
+yup-oauth2 = { version = "9.0.0", optional = true }
```
Comment (lalitb) [inline opentelemetry-stackdriver/Cargo.toml:26]: build at root level fails without this upgrade.
```
error: failed to select a version for the requirement `yup-oauth2 = "^8.3.3"`
candidate versions found which didn't match: 9.0.0, 8.3.2, 8.3.1, ...
```

File: opentelemetry-user-events-logs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+## v0.4.0
+
+### Changed
+
+- Bump opentelemetry version to 0.23, opentelemetry_sdk version to 0.23
```
Comment (cijothomas) [inline opentelemetry-user-events-logs/CHANGELOG.md:9]:
```suggestion
- Bump opentelemetry, opentelemetry_sdk  version to 0.23.
```

File: opentelemetry-stackdriver/Cargo.toml (hunk 1 above)
Comment (cijothomas) [reply opentelemetry-stackdriver/Cargo.toml:26]: lets do it as a separate PR.

**Comment by lalitb** [comment]
closing this. Will open two separate PRs - one for otel-rust dependency upgrade, and other for user_events exporter release.

**Pull request closed without merging**